src/libmtl/include/libmtl/MTLFormula.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+#include "Interval.h"
+
+#include <algorithm>
+#include <cassert>
+#include <optional>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+namespace logic {
+
+using TimePoint    = int;
+using TimeInterval = ::arithmetic::Interval<TimePoint>;
+
+class MTLFormula;
+
+/**
+ * @brief Strong typing of atomic propositions, allows simplified construction from Boolean values
+ * as well.
+ *
+ */
+struct AtomicProposition
+{
+	AtomicProposition() = delete;
+
+	AtomicProposition(const std::string &name) : ap_(name)
+	{
+	}
+
+	template <typename Y, typename T = std::enable_if_t<std::is_same<Y, bool>{}>>
+	AtomicProposition(Y b)
+	{
+		ap_ = b ? "true" : "false";
+	}
```

Comment:
Why is this a template? It looks like this is used to allow implicit conversion from a bool while not using that overload for a string, is that correct?

Reply:
Yes, that is the idea. String literals such as "0" will be converted (implicitly) to Boolean, which I wanted to avoid.